File: 1950-sign-of-the-product-of-an-array/1950-sign-of-the-product-of-an-array.js
var arraySign = function(nums) {
    let countNeg = 0;
    let countZero = 0;

    for (let num of nums) {
        if (num < 0) {
            countNeg--;
        } else if (num === 0) {
            countZero++;
        }
    }

    if (countZero >= 1) {
        return 0;
    }

    return countNeg % 2 === 0 ? 1 : -1;
};

File: 1950-sign-of-the-product-of-an-array/1950-sign-of-the-product-of-an-array.cs
public class Solution {
    public int ArraySign(int[] nums) {
        int countNeg = 0;
        int countZero = 0;

        foreach (int num in nums) {
            if (num < 0) {
                countNeg--;
            } else if (num == 0) {
                countZero++;
            }
        }

        if (countZero >= 1) {
            return 0;
        }

        return countNeg % 2 == 0 ? 1 : -1;
    }
}